### ISSUE TITLE
DialogueRunner's OnNodeComplete() not running 

### ISSUE THREAD

**Issue by zoramikau** [YarnSpinnerTool/YarnSpinner#270]
Hey YarnSpinner team, 

I'm rather new to this so it's possible this is more of a feature, not a bug, but I'm having an issue in my YarnSpinner script where OnNodeComplete() isn't running for a function attached to the Dialogue Runner. 

My script looks like this when it doesn't run: 

`title: Vanilla
tags:
---
<<if visited("Vanilla") is false>>
    Nightshade: Hey, Vanilla.
    Vanilla: Glad to see you back. 
    Nightshade: Anything I can do to help? Things feel out of sorts.
    Vanilla: Of course! If you could get us some more wood it'd be appreciated.
    Vanilla: About 20 pieces should do. 
<<else>>
    Vanilla: Hey!
<<endif>>

<<if visited("Vanilla") == true && inventory_count("Wood", 20) == false>>
    Vanilla: Still waiting on that wood. 
    Vanilla: Remember: 20 pieces should be enough.
    Vanilla: Thanks for the help! 
<<endif>>

<<if visited("Vanilla") == true && inventory_count("Wood", 20) == true>>
    Vanilla: You got it. 
    Vanilla: I appreciate it a lot.
    [[Anything else I can do?|AnythingElse]]
    [[Glad I could help!|Vanilla.GladHelp]]
<<endif>>
===

title: AnythingElse
tags: 
---
Vanilla: Not right now. I'll let you know. Thanks again!
===

title: Vanilla.GladHelp
tags: 
---
Vanilla: I'll let you know if I need anything else! Thanks again!
===`

But if I remove the second two if statements from the "Vanilla" node OnNodeComplete() runs fine. 

I'm using YarnSpinner 1.2.7 and Unity 2019.4.3f1 Personal. 

Let me know if there is any more information you need, or formatting you'd like me to try. Even if this is how it's intended to work I'm not getting an error or anything telling me something is wrong and that by itself would be a useful feature.

**Comment by desplesda** [comment]
Yes, that should be called at the end of every node regardless of how the node is exited, as far as I'm aware! Will take a look.
